dev/docker/doris/start_be.sh

```diff
@@ -0,0 +1,307 @@
+set_tcmalloc_heap_limit() {
+    local total_mem_mb
+    local mem_limit_str
+
+    if [[ "$(uname -s)" != 'Darwin' ]]; then
+        total_mem_mb="$(free -m | grep Mem | awk '{print $2}')"
+    else
+        total_mem_mb="$(($(sysctl -a hw.memsize | awk '{print $NF}') / 1024))"
+    fi
+    mem_limit_str=$(grep ^mem_limit "${DORIS_HOME}"/conf/be.conf)
+    local digits_unit=${mem_limit_str##*=}
+    digits_unit="${digits_unit#"${digits_unit%%[![:space:]]*}"}"
+    digits_unit="${digits_unit%"${digits_unit##*[![:space:]]}"}"
+    local digits=${digits_unit%%[^[:digit:]]*}
+    local unit=${digits_unit##*[[:digit:] ]}
+
+    mem_limit_mb=0
+    case ${unit} in
+    t | T) mem_limit_mb=$((digits * 1024 * 1024)) ;;
+    g | G) mem_limit_mb=$((digits * 1024)) ;;
+    m | M) mem_limit_mb=$((digits)) ;;
+    k | K) mem_limit_mb=$((digits / 1024)) ;;
+    %) mem_limit_mb=$((total_mem_mb * digits / 100)) ;;
+    *) mem_limit_mb=$((digits / 1024 / 1024 / 1024)) ;;
+    esac
+
+    if [[ "${mem_limit_mb}" -eq 0 ]]; then
+        mem_limit_mb=$((total_mem_mb * 90 / 100))
+    fi
+
+    if [[ "${mem_limit_mb}" -gt "${total_mem_mb}" ]]; then
+        echo "mem_limit is larger than whole memory of the server. ${mem_limit_mb} > ${total_mem_mb}."
+        return 1
+    fi
+    export TCMALLOC_HEAP_LIMIT_MB=${mem_limit_mb}
+}
+
+# set_tcmalloc_heap_limit || exit 1
+
+## set hdfs3 conf
+if [[ -f "${DORIS_HOME}/conf/hdfs-site.xml" ]]; then
+    export LIBHDFS3_CONF="${DORIS_HOME}/conf/hdfs-site.xml"
+fi
+
+# check java version and choose correct JAVA_OPTS
+java_version="$(
+    set -e
+    jdk_version "${JAVA_HOME}/bin/java"
+)"
+
+CUR_DATE=$(date +%Y%m%d-%H%M%S)
+LOG_PATH="-DlogPath=${DORIS_HOME}/log/jni.log"
+COMMON_OPTS="-Dsun.java.command=DorisBE -XX:-CriticalJNINatives"
+JDBC_OPTS="-DJDBC_MIN_POOL=1 -DJDBC_MAX_POOL=100 -DJDBC_MAX_IDEL_TIME=300000 -DJDBC_MAX_WAIT_TIME=5000"
+
+if [[ "${java_version}" -gt 8 ]]; then
+    if [[ -z ${JAVA_OPTS_FOR_JDK_9} ]]; then
+        JAVA_OPTS_FOR_JDK_9="-Xmx1024m ${LOG_PATH} -Xlog:gc:${DORIS_HOME}/log/be.gc.log.${CUR_DATE} ${COMMON_OPTS} ${JDBC_OPTS}"
+    fi
+    final_java_opt="${JAVA_OPTS_FOR_JDK_9}"
+else
+    if [[ -z ${JAVA_OPTS} ]]; then
+        JAVA_OPTS="-Xmx1024m ${LOG_PATH} -Xloggc:${DORIS_HOME}/log/be.gc.log.${CUR_DATE} ${COMMON_OPTS} ${JDBC_OPTS}"
+    fi
+    final_java_opt="${JAVA_OPTS}"
+fi
+
+if [[ "${MACHINE_OS}" == "Darwin" ]]; then
+    max_fd_limit='-XX:-MaxFDLimit'
+
+    if ! echo "${final_java_opt}" | grep "${max_fd_limit/-/\-}" >/dev/null; then
+        final_java_opt="${final_java_opt} ${max_fd_limit}"
+    fi
+
+    if [[ -n "${JAVA_OPTS}" ]] && ! echo "${JAVA_OPTS}" | grep "${max_fd_limit/-/\-}" >/dev/null; then
+        JAVA_OPTS="${JAVA_OPTS} ${max_fd_limit}"
+    fi
+fi
+
+# set LIBHDFS_OPTS for hadoop libhdfs
+export LIBHDFS_OPTS="${final_java_opt}"
+
+#echo "CLASSPATH: ${CLASSPATH}"
+#echo "LD_LIBRARY_PATH: ${LD_LIBRARY_PATH}"
+#echo "LIBHDFS_OPTS: ${LIBHDFS_OPTS}"
```

Comment:
If these codes are invalidated, I think better to remove them.

Reply:
`start_be.sh` is a shell script copied from Apache Doris packages `bin/start_be.sh`,

There are some code like this:

```
if [[ "$(uname -s)" != 'Darwin' ]]; then
    MAX_MAP_COUNT="$(cat /proc/sys/vm/max_map_count)"
    if [[ "${MAX_MAP_COUNT}" -lt 2000000 ]]; then
        echo "Please set vm.max_map_count to be 2000000 under root using 'sysctl -w vm.max_map_count=2000000'."
        exit 1
    fi
fi

MAX_FILE_COUNT="$(ulimit -n)"
if [[ "${MAX_FILE_COUNT}" -lt 65536 ]]; then
    echo "Please set the maximum number of open file descriptors to be 65536 using 'ulimit -n 65536'."
    exit 1
fi
```

it will check `vm.max_map_count`. 
Also can be seen on https://doris.apache.org/docs/install/construct-docker/run-docker-cluster/#special-case-description

I think it's not necessary for Doris IT Environment.

So, I have removed this snippet of code, but retained the Apache Doris License.

Do you have any better suggestions?